fmriprep/workflows/bold/base.py

```diff
@@ -146,6 +147,11 @@ def init_func_preproc_wf(bold_file, ignore, freesurfer,
         template_out_grid : str
             Keyword ('native', '1mm' or '2mm') or path of custom reference
             image for normalization
+        keep_non_denoised : bool
```

Comment:
This language might get a little confusing as we're also integrating tedana denoising. Should we clarify if AROMA vs tedana denoising was used, rather than just "non-denoised", generally ?

Reply:
I agree, that will get confusing. So this could be `keep_aroma_non_denoised`?

And then other variables with `denoised` with have `aroma` inserted into their names so that will not lead to conflict.

Is there an opinion on any shortening of the word denoised that would still be clear? (just so the variables don't become overly long, but still descriptive)
1. `dnsd`
2. `densd`
3. `dnsed`

Reply:
Are there situations in which someone will perform both tedana and AROMA denoising? And if not, would you want to keep a variant that is denoised by one and no the other?

If not, this could be a generic flag.

Reply:
I'd argue that we should make denoising by tedana and AROMA mutually exclusive, since at least tedana will not function properly if performed after AROMA denoising

In the case of tedana, then, this would be keeping the optimal combination in addition to the denoised time series ? We do this by default, but I'm fine to make it optional :+1:

Reply:
That seems fine to me. Am I correct in understanding "optimal combination" to be a synthetic BOLD series that estimates the true signal as closely as possible from the multiple echoes?

I think there are two (potentially) competing impulses here. The first is to give people the outputs they're going to use and not (especially if large) those they won't, while the second is to provide consistent behavior when analogous algorithms are applied.

The assumption with AROMA is that people want the non-aggressively denoised variant, and aren't going to bother with the standard preprocessed BOLD series except in exploratory situations. I'm not sure what the equivalent assumptions are with tedana. If they're too different, we can break the second rule, but my guess is that you'll generally want one or the other, so it'll mostly be a question of what's the default and what the best way of indicating that is. (e.g. perhaps we want `--tedana=combined` and `--tedana=denoised`)

Getting fairly off-topic for this PR, though. Sorry.